workbench-core/authorization/src/caslAuthorizationPlugin.test.ts

```diff
@@ -154,4 +155,94 @@ describe('CASL Authorization Plugin', () => {
       }
     );
   });
+
+  describe('isAuthorizedOnDynamicOperations', () => {
+    let mockIdentityPermissions: IdentityPermission[];
+    beforeEach(() => {
+      mockIdentityPermissions = [
+        {
+          effect: 'ALLOW',
+          action: 'CREATE',
+          identityType: 'GROUP',
+          identityId: 'SampleGroup',
+          subjectType: 'SampleSubject',
+          subjectId: '*',
+          conditions: { env: { $eq: 'SampleEnv' } }
+        },
+        {
+          effect: 'ALLOW',
+          action: 'UPDATE',
+          identityType: 'GROUP',
+          identityId: 'SampleGroup',
+          subjectType: 'SampleSubject',
+          subjectId: '1234'
+        },
+        {
+          effect: 'DENY',
+          action: 'UPDATE',
+          identityType: 'GROUP',
+          identityId: 'SampleGroup',
+          subjectType: 'SampleSubject',
+          subjectId: '1234'
+        }
+      ];
+      caslAuthorizationPlugin = new CASLAuthorizationPlugin();
+    });
+    test('Check for valid dynamic operations with valid conditions', async () => {
+      const mockDynamicOperations: DynamicOperation[] = [
+        {
+          action: 'CREATE',
+          subjectType: 'SampleSubject',
+          subjectId: '1',
+          subjectAttributes: {
+            env: 'SampleEnv'
+          }
+        }
+      ];
+      const response = await caslAuthorizationPlugin.isAuthorizedOnDynamicOperations(
+        mockIdentityPermissions,
+        mockDynamicOperations
+      );
+      expect(response).toBeUndefined();
+    });
+
+    test('Check for a invalid dynamic operations due invalid conditions', async () => {
+      const mockDynamicOperations: DynamicOperation[] = [
+        {
+          action: 'CREATE',
+          subjectType: 'SampleSubject',
+          subjectId: '1'
+        }
+      ];
+      try {
+        await caslAuthorizationPlugin.isAuthorizedOnDynamicOperations(
+          mockIdentityPermissions,
+          mockDynamicOperations
+        );
+        expect.hasAssertions();
+      } catch (err) {
+        expect(err).toBeInstanceOf(ForbiddenError);
```

Comment:
What about using child classes for the errors to make them less generic to their context? This one could be something like `CreateForbiddenError` which inherits from `ForbiddenError`. Then we only need to check the right type is being thrown and don't need to do any matching on the message (which could change as we keep working on the project).

Reply:
I can see this as an option, but one thing to consider is that this doesn't scale well if we change `action`. We can move in the direction of using CRUD + other actions then we need errors for each action failed. I can see a scenario where consumers want to handle a specific action error but is that a valid use case right now?

Reply:
Not a deal breaker for now, but maybe something we consider in the future to simplify error management.